app/forms/declaration/CommodityMeasure.scala

```diff
@@ -61,13 +84,34 @@ object CommodityMeasure extends DeclarationPage {
         .verifying("declaration.commodityMeasure.grossMass.error", isEmpty or validateDecimalGreaterThanZero(16)(6) and containsNotOnlyZeros)
     ),
     "netMass" -> optional(
-      text()
-        .verifying("declaration.commodityMeasure.netMass.error", isEmpty or validateDecimalGreaterThanZero(16)(6) and containsNotOnlyZeros)
+      of(
+        CrossFieldFormatter(
+          secondaryKey = "grossMass",
+          constraints = Seq(
+            ("declaration.commodityMeasure.netMass.error.format", (net: String, _: String) => netMassFormatValidation(net)),
+            (
+              "declaration.commodityMeasure.netMass.error.biggerThanGrossMass",
+              (net: String, gross: String) => !netMassFormatValidation(net) || isFirstGreaterOrEqual(gross, net)
+            )
+          )
+        )
+      )
     )
   )(CommodityMeasure.applyClearance)(CommodityMeasure.unapplyClearance)
 
+  private def isFirstGreaterOrEqual(first: String, second: String): Boolean = {
+    val firstNumOpt = Try(BigDecimal(first)).toOption
+    val secondNumOpt = Try(BigDecimal(second)).toOption
+
+    (for {
+      firstNum <- firstNumOpt
+      secondNum <- secondNumOpt
+    } yield firstNum >= secondNum).getOrElse(true)
```

Comment:
If the second will be defined and first not defined you will have a true here. Is that correct? or Is that possible?

Reply:
It is possible and it is correct in this case, but the wording could be probably a little bit better.